app/actions/service_credential_binding_app_create.rb

```diff
@@ -50,16 +48,17 @@ def precursor(service_instance, app: nil, volume_mount_services_enabled: false,
 
       private
 
-      def validate!(service_instance, app, volume_mount_services_enabled)
+      def validate!(service_instance, app, volume_mount_services_enabled, binding)
         app_is_required! unless app.present?
         space_mismatch! unless all_space_guids(service_instance).include? app.space.guid
-
         if service_instance.managed_instance?
           service_not_bindable! unless service_instance.service_plan.bindable?
           service_not_available! unless service_instance.service_plan.active?
           volume_mount_not_enabled! if service_instance.volume_service? && !volume_mount_services_enabled
           operation_in_progress! if service_instance.operation_in_progress?
         end
+        already_bound! if binding && (binding.create_succeeded? || binding.create_in_progress?)
```

Comment:
Maybe this would be slightly more readable:
```
if binding
  already_bound! if binding.create_succeeded? || binding.create_in_progress?
  ...
end
```